rpcserver.go

```diff
@@ -1969,39 +2057,26 @@ func (r *rpcServer) AbandonChannel(ctx context.Context,
 	index := in.ChannelPoint.OutputIndex
 	chanPoint := wire.NewOutPoint(txid, index)
 
-	// With the chanPoint constructed, we'll attempt to find the target
-	// channel in the database. If we can't find the channel, then we'll
-	// return the error back to the caller.
-	dbChan, err := r.server.chanDB.FetchChannel(*chanPoint)
+	// Abandoning a channel is a three step process: remove from the open
+	// channel state, remove from the graph, remove from the contract
+	// court. Between any step it's possible that the users restarts the
+	// process all over again. As a result, each of the steps below are
+	// intended to be idempotent.
+	err = abandonChanFromDatabase(
+		r.server.chanDB, r.server.cc.chainIO, chanPoint,
+	)
 	if err != nil {
 		return nil, err
 	}
-
-	// Now that we've found the channel, we'll populate a close summary for
-	// the channel, so we can store as much information for this abounded
-	// channel as possible. We also ensure that we set Pending to false, to
-	// indicate that this channel has been "fully" closed.
-	_, bestHeight, err := r.server.cc.chainIO.GetBestBlock()
+	err = abandonChanFromGraph(
```

Comment:
Maybe dumb question due to my unfamiliarity with the codebase, but what happens if the abandonChanFromDatabase suceeds but abandonChanFromGraph fails?

Reply:
Great question! If `abandonChanFromGraph` the RPC call will return with an error. At that point, depending on the nature of the failure, the caller can re-try and essentially have the logic pick up where it left off. 

For those that have already used the existing version of this call, the first abandonment phase will be a noop, and they'll move onto removing things from the channel graph.